### PR TITLE
Changes welding mask to use an overlay instead of a tint

### PR DESCRIPTION
**Pull request by TobleroneSwordfish**
[BALANCE]
## About the PR 
Remake of #9398 based on feedback and Opinions
Vision range is larger than blinded, but still enough to be a major hinderence.
Also makes the overlay update instantly.
image

## Why's this needed? 
Using screen tint in game balance is a bad idea, vision obscuring overlays seem like a better way to go.

## Changelog

```changelog
(u)LeahTheTech
(+)Welding masks now use a vision obscuring overlay instead of a screen tint
```

### CODE REVIEW

**Review by flrsh — APPROVED** [review]
Looks good to me.

File: code/obj/item/clothing/helmets.dm [hunk 1]
```diff
@@ -687,15 +690,28 @@
...
+
+	unequipped(mob/user)
+		. = ..()
+		src.reveal(user)
+
+	disposing()
+		. = ..()
+		if (ismob(src.loc))
+			var/mob/owner = src.loc
+			src.reveal(owner)
```
Comment (Tarmunora) [inline code/obj/item/clothing/helmets.dm:713]: hm. If you're wearing a weldingmask, and also have one in your hand, would pouring acid on the one in your hand clear the overlay b/c of this? Might be worth testing

Comment (Azrun) [reply code/obj/item/clothing/helmets.dm:713]: This. Likely checks for if it is in slot.

**Review by MarkNais — APPROVED** [review]
Pending Tarm's comment, looks good to me.
There used to be an old bug involving wearing mesons and the welding helmet at the same time, and toggling the mesons on/off and the helmet up/down thru different combinations. I'm doubtful this code will bring it back, but please test wearing mesons when you do the acid test.
(Also wondering if you'll "see thru walls" with both meson and helmet - I'm doubtful given the overlay)

**Comment by TobleroneSwordfish** [comment]
Good catch, welding masks in hand now cannot update the overlay. Mesons seem to work as expected.

**Review by ZeWaka — APPROVED** [review]

**Comment by ZeWaka** [comment]
The council has spoken, I suppose.

**Pull request merged**